sources/gdal/large_image_source_gdal/__init__.py

```diff
@@ -261,7 +261,7 @@ def _setDefaultStyle(self):
             styleBands = self.style['bands'] if 'bands' in self.style else [self.style]
             if not len(styleBands) or (len(styleBands) == 1 and isinstance(
                     styleBands[0].get('band', 1), int) and styleBands[0].get('band', 1) <= 0):
-                del self.style
+                del self._style
```

Comment:
Also change `self.style = ...` to `self._style = ...` in this method (currently on line 294) and a similar `self.style = ...` in the mapnik source (line 168).